Source/Runtime/Events.Processing/Projections.proto

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+syntax = "proto3";
+
+import "Fundamentals/Artifacts/Artifact.proto";
+import "Fundamentals/Protobuf/Uuid.proto";
+import "Fundamentals/Protobuf/Failure.proto";
+import "Fundamentals/Services/ReverseCallContext.proto";
+import "Fundamentals/Services/Ping.proto";
+import "Runtime/Events.Processing/StreamEvent.proto";
+import "Runtime/Events.Processing/Processors.proto";
+
+package dolittle.runtime.events.processing;
+
+option csharp_namespace = "Dolittle.Runtime.Events.Processing.Contracts";
+
+enum ProjectEventKeySelectorType {
+    EVENT_SOURCE_ID = 0;
+    PARTITION_ID = 1;
+    PROPERTY = 2;
+}
+
+message ProjectionEventSpecification {
+    artifacts.Artifact eventType = 1;
+    ProjectEventKeySelectorType keySelectorType = 2;
+    string keySelectorExpression = 3;
+}
+
+message ProjectionRegistrationRequest {
+    services.ReverseCallArgumentsContext callContext = 1;
+    protobuf.Uuid scopeId = 2;
+    protobuf.Uuid projectionId = 3;
+    repeated ProjectionEventSpecification events = 4;
+}
+
+message ProjectionResponse {
+    services.ReverseCallResponseContext callContext = 1;
+    string nextState = 2;
+    ProcessorFailure failure = 3; // If not set/empty - no failure
+}
+
+message ProjectionClientToRuntimeMessage {
+    oneof Message {
+        ProjectionRegistrationRequest registrationRequest = 1;
+        ProjectionResponse handleResult = 2;
+        services.Pong pong = 3;
+    }
+}
+
+message ProjectionRegistrationResponse {
+    protobuf.Failure failure = 1;
+}
+
+message ProjectionRequest {
+    services.ReverseCallRequestContext callContext = 1;
+    string previousState = 2;
+    StreamEvent event = 3;
+    RetryProcessingState retryProcessingState = 4;
+}
+
+message ProjectionRuntimeToClientMessage {
+    oneof Message {
+        ProjectionRegistrationResponse registrationResponse = 1;
+        ProjectionRequest handleRequest = 2;
```

Comment:
Same as above; 'projectionRequest' ? But now that I see this now and think about it 'handleResult' and 'handleRequest' does make somewhat sense as well.

Reply:
Yes, this was copied from the EventHandlers which had the Request and Response names - I kind of liked it. In terms of the protocol, there's a request Runtime -> Head, and for each request a response Head -> Runtime.